packages/core/src/system.ts

```diff
@@ -262,3 +341,26 @@ export type InspectionEvent =
   | InspectedSnapshotEvent
   | InspectedEventEvent
   | InspectedActorEvent;
+
+export interface ActorRegisteredEvent<
+  TActorRef extends AnyActorRef,
+  TSystemId extends string = string
```

Comment:
This made sense to me if there were any plans to allow for extracting `systemId` types from a machine.

Let me if it doesn't!